src/client/Client.cc

```diff
@@ -6011,6 +6012,9 @@ void Client::tick()
     }
   }
 
+  if (mounted && !mds_requests.empty() && now - curr_req_stamp > 120)
+    objecter->get_latest_version(0, curr_req_stamp, NULL);
```

Comment:
While the patch fixes the issue, there is, at least, one technical issue: the second parameter is incorrect. It is supposed to be of the type `epoch_t`. Client.cc has only 2 variables that are supposed to hold epoch values: `cap_epoch_barrier` and `local_osd_epoch`. But both of them hold the value `0` when the client has hanged. So passing any of them wouldn't update client's OSD map.